plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/pylint/PyLintAnalysisTest.java

```diff
@@ -0,0 +1,92 @@
+                        "        if letter == ' ':\n" +
+                        "            encoded = encoded + ' '\n" +
+                        "        else:\n" +
+                        "            x = letters.index(letter) + shift\n" +
+                        "            encoded=encoded + letters[x]\n" +
+                        "if choice == \"decode\":\n" +
+                        "    for letter in word:\n" +
+                        "        if letter == ' ':\n" +
+                        "            encoded = encoded + ' '\n" +
+                        "        else:\n" +
+                        "            x = letters.index(letter) - shift\n" +
+                        "            encoded = encoded + letters[x]\n" +
+                        "print(encoded)");
+
+        PyLintAnalysis pyLintAnalysis = new PyLintAnalysis(null, document, null, new NullProgressMonitor(),
+                null);
+
+        String output = "************* Module snippet\n" +
+                "C: 14, 0: (bad-whitespace) Exactly one space required around assignment\n" +
+                "            encoded=encoded + letters[x]\n" +
+                "\n" +
+                "                   ^\n" +
+                "C: 22, 0: (missing-final-newline) Final newline missing\n" +
+                "C: 22, 0: (superfluous-parens) Unnecessary parens after 'print' keyword\n" +
+                "C:  1, 0: (missing-docstring) Missing module docstring\n" +
+                "C:  3, 0: (invalid-name) Constant name \"shift\" doesn't conform to UPPER_CASE naming style\n" +
+                "C:  4, 0: (invalid-name) Constant name \"choice\" doesn't conform to UPPER_CASE naming style\n" +
+                "C:  5, 0: (invalid-name) Constant name \"word\" doesn't conform to UPPER_CASE naming style\n" +
+                "C:  6, 0: (invalid-name) Constant name \"letters\" doesn't conform to UPPER_CASE naming style\n" +
+                "C:  7, 0: (invalid-name) Constant name \"encoded\" doesn't conform to UPPER_CASE naming style\n" +
+                "\n" +
+                "------------------------------------------------------------------\n" +
+                "\n" +
+                "Your code has been rated at 5.26/10 (previous run: 5.26/10, +0.00)\n";
+
+        PydevPrefs.getEclipsePreferences().putInt(PyLintPreferences.SEVERITY_WARNINGS, IMarker.SEVERITY_ERROR);
+        PydevPrefs.getEclipsePreferences().putInt(PyLintPreferences.SEVERITY_CODING_STANDARD, IMarker.SEVERITY_ERROR);
+        PydevPrefs.getEclipsePreferences().putInt(PyLintPreferences.SEVERITY_REFACTOR, IMarker.SEVERITY_ERROR);
+
+        pyLintAnalysis.afterRunProcess(output, "No config file found, using default configuration\n", null);
+
+        List<String> markersMessage = new ArrayList<String>();
+
+        for (MarkerInfo marker : pyLintAnalysis.markers) {
+            markersMessage.add(marker.message);
+        }
+
+        assertEquals("PyLint: bad-whitespace", markersMessage.get(0));
```

Comment:
This doesn't seem right.

This should be the `PyLintVisitor.PYLINT_MESSAGE_ID` and not the message (you should check both and also the line).

The error is happening because you switched the regexp groups order but didn't fix the groups used in the code afterwards.

i.e. the groups must be updated in:
```
                        Matcher m = PYLINT_MATCH_PATTERN.matcher(tok);
                        if (m.matches()) {
                            line = Integer.parseInt(tok.substring(m.start(1), m.end(1)));
                            column = Integer.parseInt(tok.substring(m.start(2), m.end(2)));
                            messageId = tok.substring(m.start(4), m.end(4)).trim();
                            tok = tok.substring(m.start(3), m.end(3)).trim();
                        } else {
                            continue;
                        }
```

Reply:
That's right, I didn't see that. I already made the fix ;)